coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/parsetree/ParseTreeBuilder.java

```diff
@@ -529,7 +530,19 @@ private SqlNode visitFunctionInternal(ASTNode node, ParseContext ctx, SqlLiteral
   protected SqlNode visitSelectExpr(ASTNode node, ParseContext ctx) {
     List<SqlNode> sqlNodes = visitChildren(node, ctx);
     if (sqlNodes.size() == 1) {
-      return sqlNodes.get(0);
+      final SqlNode sqlNode = sqlNodes.get(0);
+      // if select a field from a struct column, like
+      // SELECT table.struct_column.field_name FROM table
+      // we need to add an alias to avoid null pointer exception:
```

Comment:
can you clarify in the comment how a null pointer is thrown if no alias is added?